specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/preview/2022-08-07/iotdps.json

```diff
@@ -0,0 +1,2602 @@
+        "keyName": {
+          "type": "string",
+          "description": "Name of the key."
+        },
+        "primaryKey": {
+          "type": "string",
+          "description": "Primary SAS key value."
+        },
+        "secondaryKey": {
+          "type": "string",
+          "description": "Secondary SAS key value."
+        },
+        "rights": {
+          "description": "Rights that this key has.",
+          "enum": [
+            "ServiceConfig",
+            "EnrollmentRead",
+            "EnrollmentWrite",
+            "DeviceConnect",
+            "RegistrationStatusRead",
+            "RegistrationStatusWrite"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "AccessRightsDescription",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "keyName",
+        "rights"
+      ]
+    },
+    "Resource": {
+      "description": "The common properties of an Azure resource.",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource identifier."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource name.",
+          "pattern": "^(?![0-9]+$)(?!-)[a-zA-Z0-9-]{2,49}[a-zA-Z0-9]$"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "location": {
+          "type": "string",
+          "description": "The resource location."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The resource tags."
+        }
+      },
+      "x-ms-azure-resource": true,
+      "required": [
+        "location"
+      ]
+    },
+    "OperationListResult": {
+      "description": "Result of the request to list provisioning service operations. It contains a list of operations and a URL link to get the next set of results.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Provisioning service operations supported by the Microsoft.Devices resource provider.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/Operation"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "URL to get the next set of operation list results if there are any."
+        }
+      }
+    },
+    "Operation": {
+      "description": "Provisioning Service REST API operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Operation name: {provider}/{resource}/{read | write | action | delete}"
+        },
+        "isDataAction": {
```

Comment:
>isDataAction [](http://example.com/codeflow?start=9&length=12)

Did you miss: `"readOnly": true` mark this property as readonly if your RP is setting it.